src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorEditorFactory.cs

```diff
@@ -0,0 +1,166 @@
+                var adaptersFactory = componentModel.GetService<IVsEditorAdaptersFactoryService>();
+                return adaptersFactory;
+            });
+
+            _razorLSPContentType = new Lazy<IContentType>(() =>
+            {
+                var componentModel = (IComponentModel)AsyncPackage.GetGlobalService(typeof(SComponentModel));
+                var contentTypeService = componentModel.GetService<IContentTypeRegistryService>();
+                var contentType = contentTypeService.GetContentType(RazorLSPContentTypeDefinition.Name);
+                return contentType;
+            });
+
+            _featureFlags = new Lazy<IVsFeatureFlags>(() =>
+            {
+                var featureFlags = (IVsFeatureFlags)AsyncPackage.GetGlobalService(typeof(SVsFeatureFlags));
+                return featureFlags;
+            });
+        }
+
+        private bool IsLSPRazorEditorEnabled
+        {
+            get
+            {
+                var lspRazorEnabledString = Environment.GetEnvironmentVariable(RazorLSPEditorFeatureFlag);
+                bool.TryParse(lspRazorEnabledString, out var enabled);
+                if (enabled)
+                {
+                    return true;
+                }
+
+                enabled = _featureFlags.Value.IsFeatureEnabled(RazorLSPEditorFeatureFlag, defaultValue: false);
+                return enabled;
+            }
+        }
+
+        public override int CreateEditorInstance(
+            uint createDocFlags,
+            string moniker,
+            string physicalView,
+            IVsHierarchy pHier,
+            uint itemid,
+            IntPtr existingDocData,
+            out IntPtr docView,
+            out IntPtr docData,
+            out string editorCaption,
+            out Guid cmdUI,
+            out int cancelled)
+        {
+            if (!IsLSPRazorEditorEnabled)
+            {
+                docView = default;
+                docData = default;
+                editorCaption = null;
+                cmdUI = default;
+                cancelled = 0;
+
+                // Razor LSP is not enabled, allow another editor to handle this document
+                return VSConstants.VS_E_UNSUPPORTEDFORMAT;
+            }
+
+            var editorInstance = base.CreateEditorInstance(createDocFlags, moniker, physicalView, pHier, itemid, existingDocData, out docView, out docData, out editorCaption, out cmdUI, out cancelled);
+            var textLines = (IVsTextLines)Marshal.GetObjectForIUnknown(docData);
+
+            SetTextBufferContentType(textLines);
+
+            return editorInstance;
+        }
+
+        private void SetTextBufferContentType(IVsTextLines textLines)
+        {
+            var textBufferDataEventsGuid = typeof(IVsTextBufferDataEvents).GUID;
+            var connectionPointContainer = textLines as IConnectionPointContainer;
+            connectionPointContainer.FindConnectionPoint(textBufferDataEventsGuid, out var connectionPoint);
```

Comment:
Does this have a return value we should check?

Reply:
Nope, void returning